test/test_cuda.py

```diff
@@ -1717,24 +1750,33 @@ def test_stream_event_nogil(self):
                           TestCuda._event_wait]:
             p2c = queue.Queue()
             c2p = queue.Queue()
+            e_tik = torch.cuda.Event(enable_timing=True)
+            e_tok = torch.cuda.Event(enable_timing=True)
 
             t = threading.Thread(
                 target=TestCuda._test_stream_event_nogil,
-                args=(sync_func, p2c, c2p))
+                args=(self, sync_func, p2c, c2p))
             t.daemon = True
             t.start()
 
             c2p.get()
-            tik = time.time()
             with torch.cuda.device('cuda:0'):
+                e_tik.record()
                 p2c.put(0)
-                sync_func(50000000)
-
-            t.join()
-            tok = time.time()
-
-            self.assertGreater(tok - tik, 0.05)
-            self.assertLess(tok - tik, 0.1)
+                parent_time = sync_func(self, 50000000)
```

Comment:
Not sure if this will make anything better, but maybe making it a global variable `FIFTY_MIL_CYCLES` helps if someone comes along and wants to modify all the places where it's used?